### PR TITLE
Adaptors task gulov

### PR DESCRIPTION
**Pull request by GIurii**

### CODE REVIEW

File: adaptors task/student/include/impl.h [hunk 1]
```diff
@@ -11,4 +13,48 @@
...
+         else
+         {
+            return false;
+         }
+      }};
+
+   for (const char& ch : source)
+   {
+      switch (ch) {
+      case '(':
```
Comment (smay1613) [inline adaptors task/student/include/impl.h:36]: USE "brackets" VECTOR FROM .cpp FILE! It is mandatory!

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: adaptors task/student/include/impl.h (hunk 1 above)
Comment (GIurii) [reply adaptors task/student/include/impl.h]: Already reworked. Please check.

File: adaptors task/student/src/impl.cpp [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
+   for (const auto& ch : source)
+   {
+      for(const auto& pair: brackets){
+         if (ch == pair.first){
+            charsStack.push(pair.second);
+         } else if (ch == pair.second) {
+            if(!charsStack.empty() && pair.second == charsStack.top()) {
+               charsStack.pop();
+            } else {
+               result = false;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:25]: return false is much simpler

File: adaptors task/student/src/impl.cpp [hunk 3]
```diff
@@ -1,8 +1,36 @@
...
+{
+   std::stack<char> charsStack;
+   bool result = true;
+
+   for (const auto& ch : source)
+   {
+      for(const auto& pair: brackets){
+         if (ch == pair.first){
+            charsStack.push(pair.second);
+         } else if (ch == pair.second) {
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:21]: please give this condition a name like
const bool isClosedBracket =

**Review by smay1613 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**